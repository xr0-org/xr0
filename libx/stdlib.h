#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
