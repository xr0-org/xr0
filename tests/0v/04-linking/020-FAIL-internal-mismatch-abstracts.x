#include <stdlib.h>

/* .x specific test */

void *
allocating() ~ [ .free(result); ];

int
main()
{
	void *p;

	p = allocating();
	free(p);
}

void *
allocating() ~ [ return .malloc(1); ]
{
	void *p;

	p = malloc(1);
	return p;
}
