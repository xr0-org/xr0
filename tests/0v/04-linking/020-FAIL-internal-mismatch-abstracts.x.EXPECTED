function `allocating' prototype and definition abstracts mismatch
